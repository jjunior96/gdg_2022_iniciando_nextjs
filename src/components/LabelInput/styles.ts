import styled, { css } from 'styled-components';

import { LabelInputProps } from '.';

type LabelContainerProps = Omit<LabelInputProps, 'label'>;

export const LabelContainer = styled.label<LabelContainerProps>`
  ${({ theme, isRequired, isDisabled, labelColor, error }) => css`
    font-size: ${theme.font.sizes.xxsmall};
    margin: 0;
    padding: 0;
    line-height: 1;

    span {
      display: none;
    }

    color: ${labelColor === 'white' ? theme.colors.white : theme.colors.black};

    ${isRequired &&
    css`
      span {
        display: inline;
        color: ${theme.colors.red};
      }
    `}

    ${isDisabled &&
    css`
      cursor: not-allowed;

      color: ${theme.colors.gray_200};
    `}

    ${error &&
    css`
      color: ${theme.colors.red};
    `}
  `}
`;
