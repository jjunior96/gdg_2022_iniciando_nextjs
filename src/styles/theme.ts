export default {
  grid: {
    container: '96rem',
    gutter: '3.2rem'
  },
  components: {
    sidebar: '20rem',
    header: '8rem'
  },
  border: {
    radius: '0.4rem'
  },
  box: {
    shadow: '0 8px 8px rgba(0, 0, 0, 0.2)'
  },
  font: {
    family:
      "Montserrat, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xxxxsmall: '0.8rem',
      xxxsmall: '1.2rem',
      xxsmall: '1.4rem',
      xsmall: '1.6rem',
      small: '2rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      xxxlarge: '5.6rem',
      huge: '6.4rem'
    }
  },
  colors: {
    primary: '#24A66B',
    secundary: '#5855E9',
    white: '#FAFAFA',
    gray_50: '#f0efeb',
    gray_100: '#D9D9D9',
    gray_200: '#91AAB4',
    gray_300: '#3E606F',
    gray_450: '#193441',
    gray_500: '#343a40',
    yellow_300: '#F1C40F',
    purple_100: '#7F57DF',
    purple_300: '#5855E9',
    green: '#24A66B',
    red: '#E63737',
    success: '#24A66B',
    error: '#E63737',
    info: '#189CEF',
    black: '#0D0D0D'
  },
  spacings: {
    xxxsmall: '0.8rem',
    xxsmall: '1rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  modal: {
    sizes: {
      small: '40rem',
      medium: '48rem',
      large: '54rem'
    }
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const;
