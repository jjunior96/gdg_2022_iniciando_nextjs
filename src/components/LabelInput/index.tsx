import * as S from './styles';

export type LabelColorType = 'white' | 'black';

export type LabelInputProps = {
  label?: string;
  htmlFor: string | undefined;
  children: React.ReactNode;
  isRequired?: boolean;
  isDisabled?: boolean;
  error?: boolean;
  labelColor?: LabelColorType;
};

const LabelInput = ({
  children,
  htmlFor = '',
  labelColor = 'black',
  isRequired,
  isDisabled,
  error = false
}: LabelInputProps) => {
  return (
    <S.LabelContainer
      htmlFor={htmlFor}
      isRequired={isRequired}
      isDisabled={isDisabled}
      labelColor={labelColor}
      error={error}
    >
      {children} <span>*</span>
    </S.LabelContainer>
  );
};

export default LabelInput;
