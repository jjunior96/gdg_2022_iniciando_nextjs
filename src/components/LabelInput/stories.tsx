import { Story, Meta } from '@storybook/react/types-6-0';

import Input from 'components/Input';

import LabelInput, { LabelInputProps } from '.';

export default {
  title: 'LabelInput',
  component: LabelInput
} as Meta;

export const Default: Story<LabelInputProps> = (args) => (
  <div
    style={{
      width: '40rem',
      margin: '0 auto',
      paddingTop: '1rem'
    }}
  >
    <form action="">
      <LabelInput {...args} />
    </form>
  </div>
);

Default.args = {
  children: 'Label'
};

export const withIput: Story<LabelInputProps> = (args) => (
  <div
    style={{
      width: '40rem',
      margin: '0 auto',
      paddingTop: '1rem'
    }}
  >
    <form action="">
      <LabelInput {...args} />
      <Input placeholder="Example" />
    </form>
  </div>
);

withIput.args = {
  label: 'input',
  children: 'Example'
};
