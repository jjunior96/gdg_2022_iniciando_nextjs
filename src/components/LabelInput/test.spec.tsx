import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/__config__/helpers';

import theme from 'styles/theme';

import LabelInput from '.';

describe('<LabelInput />', () => {
  it('should render the label default', () => {
    const { container } = renderWithTheme(
      <form>
        <LabelInput htmlFor="label-test">label</LabelInput>
        <input type="text" id="label-test" />
      </form>
    );

    const labelElement = screen.getByText(/label/i);

    expect(labelElement).toBeInTheDocument();

    expect(labelElement).toHaveStyle({
      color: theme.colors.black
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the label with isRequired props passed', () => {
    renderWithTheme(
      <form>
        <LabelInput htmlFor="label-test" isRequired>
          label
        </LabelInput>
        <input type="text" id="label-test" />
      </form>
    );

    const spanElement = screen.getByText('*');

    expect(spanElement).toHaveStyle({
      display: 'inline',
      color: theme.colors.red
    });
  });

  it('should render the label with isDisable props passed', () => {
    renderWithTheme(
      <form>
        <LabelInput htmlFor="label-test" isDisabled>
          label
        </LabelInput>
        <input type="text" id="label-test" />
      </form>
    );

    const labelElement = screen.getByText(/label/i);

    expect(labelElement).toHaveStyle({
      cursor: 'not-allowed',
      color: theme.colors.gray_200
    });
  });

  it('should render the label with labelColor white props passed', () => {
    renderWithTheme(
      <form>
        <LabelInput htmlFor="label-test" labelColor="white">
          label
        </LabelInput>
        <input type="text" id="label-test" />
      </form>
    );

    const labelElement = screen.getByText(/label/i);

    expect(labelElement).toHaveStyle({
      color: theme.colors.white
    });
  });
});
