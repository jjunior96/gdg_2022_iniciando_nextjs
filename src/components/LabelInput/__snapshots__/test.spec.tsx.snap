// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelInput /> should render the label default 1`] = `
.c0 {
  font-size: 1.4rem;
  margin: 0;
  padding: 0;
  line-height: 1;
  color: #0D0D0D;
}

.c0 span {
  display: none;
}

<form>
  <label
    class="c0"
    for="label-test"
  >
    label
     
    <span>
      *
    </span>
  </label>
  <input
    id="label-test"
    type="text"
  />
</form>
`;
